storage/innobase/handler/ha_innodb.cc

```diff
@@ -335,7 +335,8 @@ thd_destructor_proxy(void *)
 	myvar->current_cond = &thd_destructor_cond;
 
 	mysql_mutex_lock(&thd_destructor_mutex);
-	srv_running = myvar;
+	my_atomic_storeptr_explicit(&srv_running, myvar,
+				    MY_MEMORY_ORDER_RELAXED);
```

Comment:
This is kind of alright, but FWICS it is just a half of the fix. Though probably it is acceptable.

A few lines below there's second assignment `srv_running = NULL`. As well as two loads in `srv0start.cc` and `srv0srv.cc` that are most probably affected by data race as well.

Reply:
I made all accesses atomical.